src/Range.php

```diff
@@ -1227,8 +1246,8 @@ private function setStartOrEnd(string $type, Node $node, int $offset): void
         switch ($type) {
             case 'start':
                 if (
-                    $this->computePosition($bp, [$this->endNode, $this->endOffset]) === 'after'
-                    || $this->startNode->getRootNode() !== $node->getRootNode()
+                    $this->startNode->getRootNode() !== $node->getRootNode()
+                    || $this->computePosition($bp, [$this->endNode, $this->endOffset]) === 'after'
```

Comment:
This is probably a bug in the spec as it states:
> If bp is after the range’s end, or if range’s root is not equal to node’s root, set range’s end to bp.

where it clearly makes more sense to compare root nodes first

Reply:
Filed as https://github.com/whatwg/dom/issues/924